src/services/etherspot.js

```diff
@@ -583,22 +588,13 @@ export class EtherspotService {
         return null;
       }
 
-      const supportedAssets = tokens.map(parseTokenListToken);
+      let supportedAssets = tokens.map(parseTokenListToken);
 
-      // add ETH if not within tokens list (most of the time since it's not a token)
-      const supportedAssetsHaveEth = supportedAssets.some(({ symbol }) => symbol === ETH);
-      if (!supportedAssetsHaveEth) {
-        // eslint-disable-next-line i18next/no-literal-string
-        const iconUrl = 'https://tokens.1inch.exchange/0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee.png';
-        supportedAssets.push({
-          address: EthersConstants.AddressZero,
-          name: 'Ethereum', // eslint-disable-line i18next/no-literal-string
-          symbol: ETH,
-          decimals: 18,
-          iconUrl,
-          iconMonoUrl: iconUrl,
-        });
-      }
+      // TODO: replace by single invocation per chain:
+      supportedAssets = appendNativeAssetIfNeeded(CHAIN.ETHEREUM, supportedAssets);
+      supportedAssets = appendNativeAssetIfNeeded(CHAIN.POLYGON, supportedAssets);
+      supportedAssets = appendNativeAssetIfNeeded(CHAIN.XDAI, supportedAssets);
+      supportedAssets = appendNativeAssetIfNeeded(CHAIN.BINANCE, supportedAssets);
```

Comment:
This part needs to be addresses in this PR or supported assets per chains (whichever merges last).